code/modules/interview/interview.dm

```diff
@@ -0,0 +1,170 @@
+	var/can_retry = CONFIG_GET(flag/panic_bunker_interview_retries)
+	log_admin_private("[key_name(denied_by)] has denied interview #[id] for [owner_ckey][!owner ? "(DC)": ""]. Retries [can_retry ? "enabled" : "disabled"].")
+	message_admins("<span class='adminnotice'>[key_name(denied_by)] has denied interview #[id] for [owner_ckey][!owner ? "(DC)": ""]. Retries [can_retry ? "enabled" : "disabled"].</span>")
+	if(can_retry)
+		addtimer(CALLBACK(GLOB.interviews, /datum/interview_manager.proc/release_from_cooldown, owner_ckey), 180 SECONDS)
+		if (owner)
+			SEND_SOUND(owner, sound('sound/effects/adminhelp.ogg'))
+			to_chat(owner, "<font color='red' size='4'><b>-- Interview Update --</b></font>" \
+				+ "\n<span class='adminsay'>Unfortunately your interview was denied. Please try submitting another questionnaire." \
+				+ " You may do this in three minutes.</span>")
+	else
+		addtimer(CALLBACK(GLOB.interviews, /datum/interview_manager.proc/give_the_boot, owner_ckey), 60 SECONDS)
+		if (owner)
+			SEND_SOUND(owner, sound('sound/effects/adminhelp.ogg'))
+			to_chat(owner, "<font color='red' size='4'><b>-- Interview Update --</b></font>" \
+				+ "\n<span class='adminsay'>Unfortunately your interview was denied. You will be removed for the round's duration." \
+				+ " You will be kicked in 60 seconds.</span>")
+
+/**
+  * Forces client to reconnect, used in the callback from approval
+  */
+/datum/interview/proc/reconnect_owner()
+	if (!owner)
+		return
+	winset(owner, null, "command=.reconnect")
+
+/**
+  * Verb for opening the existing interview, or if relevant creating a new interview if possible.
+  */
+/mob/dead/new_player/proc/open_interview()
+	set name = "Open Interview"
+	set category = "Interview"
+	var/mob/dead/new_player/M = usr
+	if (M?.client?.interviewee)
+		var/datum/interview/I = GLOB.interviews.interview_for_client(M.client)
+		if (I) // we can be returned nothing if the user is on cooldown
+			I.ui_interact(M)
+		else
+			to_chat(usr, "<span class='adminsay'>You are on cooldown for interviews. Please wait at least 3 minutes before starting a new questionnaire.</span>")
+
+/datum/interview/ui_interact(mob/user, datum/tgui/ui = null)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "Interview")
+		ui.open()
+
+/datum/interview/ui_state(mob/user)
+	return GLOB.new_player_state
+
+/datum/interview/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	if (..())
+		return
+	switch(action)
+		if ("update_answer")
+			if (!read_only)
+				responses[text2num(params["qidx"])] = copytext_char(params["answer"], 1, 501) // byond indexing moment
+				. = TRUE
+		if ("submit")
+			if (!read_only)
+				read_only = TRUE
+				GLOB.interviews.enqueue(src)
+				. = TRUE
+		if ("approve")
+			if (usr.client?.holder && status == INTERVIEW_PENDING)
+				src.approve(usr)
+				. = TRUE
+		if ("deny")
+			if (usr.client?.holder && status == INTERVIEW_PENDING)
+				src.deny(usr)
+				. = TRUE
```

Comment:
```suggestion
				return = TRUE
```
no need to set . to true if you are not going to do anything else in the proc

Reply:
```suggestion
				return TRUE
```